### PR TITLE
Added a function to indicate deprecations

### PR DESCRIPTION
**Pull request by acrobat**
This way we can add deprecation notices. Users of 1.x will know what features/methods/locales are deprecated and removed in 2.0. This will make the upgrade a bit easier as they can change their code usages beforehand.

The result:
deprecation

### CODE REVIEW

**Comment by Azaret** [comment]
👍 Don't forget #1856

File: js/bootstrap-datepicker.js [hunk 1]
```diff
@@ -2034,3 +2038,13 @@
 	});
 
 }));
+
+function deprecate(msg) {
+	var console = window.console;
+	if (console && console.warn) {
+		console.warn('DEPRECATED: ' + msg);
```
Comment (vsn4ik) [inline js/bootstrap-datepicker.js]: Spaces with tabs!

File: js/bootstrap-datepicker.js [hunk 2]
```diff
@@ -2034,3 +2038,13 @@
 	});
 
 }));
+
+function deprecate(msg) {
```
Comment (vsn4ik) [inline js/bootstrap-datepicker.js]: Global scope is bad

Comment (acrobat) [inline js/bootstrap-datepicker.js]: How should I declare this function?

File: js/bootstrap-datepicker.js (hunk 1 above)
Comment (acrobat) [inline js/bootstrap-datepicker.js]: I leave it for now as this whole file is a mixup of spaces vs tabs. But we will clean this up and move to spaces when we move to 2.0

File: js/bootstrap-datepicker.js (hunk 2 above)
Comment (Azaret) [inline js/bootstrap-datepicker.js]: Or we just copy paste it for now in locale...

Comment (Azaret) [inline js/bootstrap-datepicker.js]: Ooorrr: `$.fn.datepicker.deprecate = function (){};`

File: js/bootstrap-datepicker.js (hunk 1 above)
Comment (vsn4ik) [inline js/bootstrap-datepicker.js]: I am sorry, my bad.

Comment (Azaret) [inline js/bootstrap-datepicker.js]: Not your fault, be sorry for the poor souls that committed mixed content... and the one that merged it 👿

File: js/bootstrap-datepicker.js (hunk 2 above)
Comment (acrobat) [inline js/bootstrap-datepicker.js]: maybe it's the best solution to move it to `$.fn.datepicker.deprecate`?

Comment (vsn4ik) [inline js/bootstrap-datepicker.js]: I like `$.fn.datepicker.deprecate`.

Comment (acrobat) [inline js/bootstrap-datepicker.js]: ok :+1: I will update the pr!

File: js/bootstrap-datepicker.js (hunk 1 above)
Comment (acrobat) [inline js/bootstrap-datepicker.js]: The one that merged it would probably be me :p

File: js/bootstrap-datepicker.js (hunk 2 above)
Comment (acrobat) [inline js/bootstrap-datepicker.js]: I've tried to move the function to `$.fn.datepicker.deprecate` but then I can't use the function in the alias method. So all tests fail, so where should I declare the method?

Comment (Azaret) [inline js/bootstrap-datepicker.js]: How? I don't see why you could not ?

Comment (acrobat) [inline js/bootstrap-datepicker.js]: I've pushed the change so you can see the test result on travis and how I declared the function!

Comment (Azaret) [inline js/bootstrap-datepicker.js]: The issue is that `alias` is executed when the datepicker is injected into jQuery.
IMO it should be executed when any of the aliases is called, not before.
I can PR you a fix for alias then you can rebase on it.

Comment (Azaret) [inline js/bootstrap-datepicker.js]: Or you should be able to do it this way:
```javascript
	function alias(method, deprecationMsg){
		return function(){
			if (typeof deprecationMsg !== 'undefined') {
				$.fn.datepicker.deprecated(deprecationMsg);
			}
			return this[method].apply(this, arguments);
		};
	}
```

Comment (acrobat) [inline js/bootstrap-datepicker.js]: yes if you want op a pr against `acrobat:deprecations`!

Comment (Azaret) [inline js/bootstrap-datepicker.js]: I don't event...
Gonna check latter how to do such thing ^^

Comment (acrobat) [inline js/bootstrap-datepicker.js]: @Azaret do you maybe have some time soon to push an update for the `deprecated` method? Or just post here a small snippet on how we should do this and I will add it!

**Comment by vsn4ik** [comment]
Thanks @acrobat!

**Pull request merged**